clarifying security patches (#3532)
dbt-labs/docs.getdbt.com: website/docs/docs/dbt-versions/core-versions.md

description: "Learn about semantic versioning for dbt Core, and how long those versions are supported."
---

dbt Core releases follow [semantic versioning](https://semver.org/) guidelines. For more on how we use semantic versions, see [How dbt Core uses semantic versioning](#how-dbt-core-uses-semantic-versioning). 

> dbt Labs provides different support levels to different versions, which may include new features, bug fixes, or security patches:

Reviewer: ```suggestion
dbt Labs provides different support levels for different versions, which may include new features, bug fixes, or security patches:
```

description: "Learn about semantic versioning for dbt Core, and how long those versions are supported."
---

dbt Core releases follow [semantic versioning](https://semver.org/) guidelines. For more on how we use semantic versions, see [How dbt Core uses semantic versioning](#how-dbt-core-uses-semantic-versioning). 

dbt Labs provides different support levels to different versions, which may include new features, bug fixes, or security patches:

- **[Active](#ongoing-patches)** &mdash; A few months after a minor version's initial release, we will provide bug fix releases to address any regressions and new bugs found in the initial release. 
- **[Critical](#ongoing-patches)** &mdash; Newer minor versions transition the previous minor version into "Critical Support" with limited "security" releases for critical security and installation fixes.
- **[End of Life](#eol-version-support)** &mdash; No longer receives bug fixes and limited to receiving only security patches. All EOL dbt Core v1.0 and later are available in dbt Cloud until further notice. 

> - **Deprecated** &mdash; Features in this state are not actively worked on or enhanced by dbt Labs.

Reviewer 2: A few notes:
- "Active" support has become a bit more active than in the past. We will patch regressions, new bugs, and even include fixes for some older bugs / quality-of-life improvements, as long as we have high confidence that those changes are narrowly scoped and will not cause unintended side effects.
- "End of Life": Minor versions that have reached EOL do not get **any** new patch releases. If it's an issue with dependencies, we resolve this within dbt Cloud. We do not resolve it by putting out a new patch release. So this sentence is not correct: "limited to receiving only security patches."
- "Deprecated": We're not talking about features here, we're talking about older releases of dbt Core. Given that the current rule applies to all pre-1.0 (v0.X) versions, should we just say that explicitly here?
> **Deprecated:** dbt-core versions older than v1.0 are no longer maintained by dbt Labs, nor supported in dbt Cloud.

Reviewer 2: Also: Should we snippetize this section, so it can be reused in the other page (without copy-paste)?

Author: thanks @jtcohen6 ! ok so to clarify the issue and [slack thread](https://dbt-labs.slack.com/archives/C0131TY7EEA/p1686693075613039?thread_ts=1686668217.878079&cid=C0131TY7EEA), which seems to focus on security patches,  is the below correct? how do we 'address this within dbt Cloud'?:

> Minor versions that have reached EOL no longer receive new patch releases. If there's an issue with security or dependencies, we address this within dbt Cloud rather than through a new patch release. 

and yea ill try to create a snippet now. i thought it wasn't going to be much work since its only two places but best to change it in one place. also the links are formatted differently depending on what page this is on but i can add the full link for all

Reviewer 2: IMO let's just cut the second sentence, and leave this as:

> Minor versions that have reached EOL no longer receive new patch releases.

We continue to make those versions available in dbt Cloud. Occasionally, that requires us to pin certain upstream dependencies while building images for older versions within the dbt Cloud application. That's our implementation detail; I don't think that's something we need to document as part of this policy.

(& totally your call on whether the snippet is worth it or not!)

description: "Learn about semantic versioning for dbt Core, and how long those versions are supported."
---

dbt Core releases follow [semantic versioning](https://semver.org/) guidelines. For more on how we use semantic versions, see [How dbt Core uses semantic versioning](#how-dbt-core-uses-semantic-versioning). 

dbt Labs provides different support levels to different versions, which may include new features, bug fixes, or security patches:
<!--

> - [For versions prior to v1.0](#version-support-prior-to-v1.0)

Author: i added this here bc i thought it would be a nice intro into the types of supported versions but commented out.

- [End of Life (EOL)](#eol-version-support)
- [Current versions](#current-version-support) (includes Minor versions, Active, and Critical)
- [Future Versions](#future-versions)
-->

| Support level | Description |
| ------------- | ----------- | 
| Active | A few months after a minor version's initial release, we will provide bug fix releases to address any regressions and new bugs found in the initial release. | 
| Critical | Newer minor versions transition the previous minor version into "Critical Support" with limited "security" releases for critical security and installation fixes. | 

> | End of Life  | No longer receive bug fixes and limited to only receiving security patches. <br /><br />All EOL dbt Core versions v1.0 and later are available in dbt Cloud until further notice. We plan to align dbt Cloud availability with dbt Core support and you'll receive advance notice before any changes take place. |

Author: ```suggestion
| End of Life  | No longer receiving bug fixes and limited to receiving only security patches. <br /><br />All EOL dbt Core versions v1.0 and later are available in dbt Cloud until further notice. We plan to align dbt Cloud availability with dbt Core support and you'll receive advance notice before any changes take place. |
```

## EOL version support 

> All dbt Core versions with an end-of-life (EOL) support level will no longer receive any fixes in new patch releases. We recommend upgrading to a newer version in [dbt Cloud](/docs/dbt-versions/upgrade-core-in-cloud) or [dbt Core](/docs/core/installation#upgrading-dbt-core) to continue receiving support. 

Reviewer: I think it's important to call out that it's bug fixes. We will provide security fixes and patches where required. We also occasionally backport non-bug fixes that address backwards compatibility. 
```suggestion
All dbt Core minor versions that have reached end-of-life (EOL) will have no new patch releases. This means they will no longer receive any fixes, including for known bugs that have been identified. Fixes for those bugs will instead be made in newer minor versions that are still under active support.

We recommend upgrading to a newer version in [dbt Cloud](/docs/dbt-versions/upgrade-core-in-cloud) or [dbt Core](/docs/core/installation#upgrading-dbt-core) to continue receiving support. 
```

Author: good question @matthewshaver, we originally had bug fixes but @jtcohen6 recommended stronger language [here](https://github.com/dbt-labs/docs.getdbt.com/pull/3532#discussion_r1234213710): "We can make this stronger: They'll no longer receive any fixes in new patch releases."

I do think including bug fixes is helpful. @jtcohen6 @matthewshaver, what about:

>All dbt Core versions with an end-of-life (EOL) support level will no longer receive any fixes, including bug fixes, in new patch releases.

Reviewer 2: That statement is accurate. Here's how I think about it:

> All dbt Core minor versions that have reached end-of-life (EOL) will have no new patch releases. This means they will no longer receive any fixes, including for known bugs that have been identified. Fixes for those bugs will instead be made in newer minor versions that are still under active support.

A lot of passive voice, and a bit redundant — but I think it's important that we drive this point home.

Author: sounds great thank you @jtcohen6 and i dont think it's redundant at all! it's very important to very clear about this, i always remember clarifying things that might seem 'obvious' to seasoned people but not so obvious to others so i always like to side with being extra clear.

## EOL version support 

> All dbt Core versions with an end-of-life (EOL) support level will no longer receive bug fixes. We recommend upgrading to a newer version in [dbt Cloud](/docs/dbt-versions/upgrade-core-in-cloud) or [dbt Core](/docs/core/installation#upgrading-dbt-core) to continue receiving bug fix support. 

Author: ```suggestion
All dbt Core versions with an end-of-life (EOL) support level will no longer receive any fixes in new patch releases. We recommend upgrading to a newer version in [dbt Cloud](/docs/dbt-versions/upgrade-core-in-cloud) or [dbt Core](/docs/core/installation#upgrading-dbt-core) to continue receiving support. 
```

## EOL version support 

> All dbt Core versions with an end-of-life (EOL) support level will no longer receive bug fixes. We recommend upgrading to a newer version in [dbt Cloud](/docs/dbt-versions/upgrade-core-in-cloud) or [dbt Core](/docs/core/installation#upgrading-dbt-core) to continue receiving bug fix support. 

Reviewer 2: > All dbt Core versions with an end-of-life (EOL) support level will no longer receive bug fixes.

We can make this stronger: They'll no longer receive any fixes in new patch releases.

All dbt Core v1.0 and later are available in dbt Cloud until further notice. In the future, we intend to align dbt Cloud availability with dbt Core ongoing support. You will receive plenty of advance notice before any changes take place.

> We have been backporting certain changes to the 1.0.latest and 1.1.latest branches of dbt-core, so that we can continue running automated tests, while those versions continue to be available in dbt Cloud.

Reviewer 2: ```suggestion
```
This is our team's choice for now. I don't think we should put it here, as part of the public policy.

> ## EOL version support 

Author: longer term, i think we should replace this eol section with a ##backporting policy header and explain our policy there.

Reviewer 2: IMO "backporting" describes an implementation detail (our git branching strategy within the `dbt-core` repo), whereas "end of life" and "ongoing support" are the intended outcome we're looking to achieve & communicate here

There should be a correspondence between "what do we (Core team) choose to backport" and "what types of fixes are included in most recent minor version (active support), versus all pre-EOL versions" — but the target audience for each feels slightly different.

> All dbt Core versions with an end-of-life (EOL) support level will no longer receive bug fixes. We recommend upgrading to a newer version in [dbt Cloud](/docs/dbt-versions/upgrade-core-in-cloud) or [dbt Core](/core/installation#upgrading-dbt-core) to continue receiving bug fix support. 

Author: ```suggestion
All dbt Core versions with an end-of-life (EOL) support level will no longer receive bug fixes.  We have been backporting certain changes to the 1.0.latest and 1.1.latest branches of dbt-core, so that we can continue running automated tests, while those versions continue to be available in dbt Cloud.

We recommend upgrading to a newer version in [dbt Cloud](/docs/dbt-versions/upgrade-core-in-cloud) or [dbt Core](/core/installation#upgrading-dbt-core) to continue receiving bug fix support. 
```

## EOL version support 

> All dbt Core versions with an end-of-life (EOL) support level will no longer receive bug fixes.  We have been backporting certain changes to the 1.0.latest and 1.1.latest branches of dbt-core, so that we can continue running automated tests, while those versions continue to be available in dbt Cloud.

Author: ```suggestion
All dbt Core versions with an end-of-life (EOL) support level will no longer receive bug fixes.  
```

We recommend upgrading to a newer version in [dbt Cloud](/docs/dbt-versions/upgrade-core-in-cloud) or [dbt Core](/docs/core/installation#upgrading-dbt-core) to continue receiving bug fix support. 

> All dbt Core versions v1.0 and later are available in dbt Cloud until further notice. In the future, we intend to align dbt Cloud availability with dbt Core ongoing support. You will receive plenty of advance notice before any changes take place.

Author: ```suggestion
All dbt Core versions v1.0 and later are available in dbt Cloud until further notice. In the future, we intend to align dbt Cloud availability with dbt Core ongoing support. You will receive plenty of advance notice before any changes take place.

We have been backporting certain changes to the 1.0.latest and 1.1.latest branches of dbt-core, so that we can continue running automated tests, while those versions continue to be available in dbt Cloud.
```
